service/app/controllers/BaseController.java

```diff
@@ -0,0 +1,225 @@
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.request.ExecutionContext;
+import org.sunbird.common.responsecode.ResponseCode;
+
+import akka.actor.ActorRef;
+import akka.actor.ActorSelection;
+import akka.pattern.Patterns;
+import akka.util.Timeout;
+import play.mvc.Result;
+import play.libs.Json;
+import play.libs.F.Function;
+import play.libs.F.Promise;
+import play.mvc.Controller;
+import play.mvc.Results;
+import play.mvc.Http.Request;
+
+/**
+ * This controller we can use for writing some common method.
+ * 
+ * @author Manzarul
+ */
+public class BaseController extends Controller {
+
+	private static final int AKKA_WAIT_TIME = 10;
+	protected Timeout timeout = new Timeout(AKKA_WAIT_TIME, TimeUnit.SECONDS);
+
+	/**
+	 * This method will make a call to Akka actor and return promise.
+	 *
+	 * @param actorRef
+	 *            ActorSelection
+	 * @param request
+	 *            Request
+	 * @param timeout
+	 *            Timeout
+	 * @param responseKey
+	 *            String
+	 * @param httpReq
+	 *            play.mvc.Http.Request
+	 * @return Promise<Result>
+	 */
+	public Promise<Result> actorResponseHandler(org.sunbird.common.request.Request request, Timeout timeout,
+			String responseKey, Request httpReq) {
+		Object actorRef = SunbirdMWService.getRequestRouter();
+
+		if (actorRef instanceof ActorRef) {
+			return Promise.wrap(Patterns.ask((ActorRef) actorRef, request, timeout))
+					.map(new Function<Object, Result>() {
+						public Result apply(Object result) {
+							if (result instanceof Response) {
+								Response response = (Response) result;
+								return createCommonResponse(request().path(), responseKey, response);
+							} else if (result instanceof ProjectCommonException) {
+								return createCommonExceptionResult(request().path(), (ProjectCommonException) result);
+							} else {
+								ProjectLogger.log("Unsupported Actor Response format", LoggerEnum.INFO.name());
+								return createCommonExceptionResult(request().path(), new Exception());
+							}
+						}
+					});
+		} else {
+			return Promise.wrap(Patterns.ask((ActorSelection) actorRef, request, timeout))
+					.map(new Function<Object, Result>() {
+						public Result apply(Object result) {
+							if (result instanceof Response) {
+								Response response = (Response) result;
+								return createCommonResponse(request().path(), responseKey, response);
+							} else if (result instanceof ProjectCommonException) {
+								return createCommonExceptionResult(request().path(), (ProjectCommonException) result);
+							} else {
+								ProjectLogger.log("Unsupported Actor Response format", LoggerEnum.INFO.name());
+								return createCommonExceptionResult(request().path(), new Exception());
+							}
+						}
+					});
+		}
+	}
+
+	public static Result createCommonExceptionResult(String path, Exception exception) {

```

Comment:
Need not be static.

Reply:
Will change static to non static later, it has reference in Global.java